### PR TITLE
[AppBar] Add new Dragons example for incorrect insets during animations

### PR DESCRIPTION
**Pull request by brianjmoore**
Adds an example case for https://github.com/material-components/material-components-ios/issues/6970

### CODE REVIEW

File: components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift [hunk 1]
```diff
@@ -0,0 +1,83 @@
+/// A simple table view controller which is manually composed.
+///
+/// This view controller is meaningfully *not* a subclass of UITableViewController and its view is
+/// not a UIScrollView subclass. This avoids logic which might attempt to be smart by detecting
+/// these cases.
+class SimpleComposedTableViewController: UIViewController {
```
Comment (romoore) [inline components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift]: Please import UIKit.

Logs:
```
components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift:6:42: error: use of undeclared type 'UIViewController'
class SimpleComposedTableViewController: UIViewController {
                                         ^~~~~~~~~~~~~~~~
components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift:36:46: error: use of undeclared type 'UITableViewDataSource'
extension SimpleComposedTableViewController: UITableViewDataSource {
                                             ^~~~~~~~~~~~~~~~~~~~~
components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift:47:46: error: use of undeclared type 'UITableViewDelegate'
extension SimpleComposedTableViewController: UITableViewDelegate {
                                             ^~~~~~~~~~~~~~~~~~~
components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift:13:34: error: use of undeclared type 'NSCoder'
  required init?(coder aDecoder: NSCoder) {
                                 ^~~~~~~
components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift:19:19: error: use of undeclared type 'MDCFlexibleHeaderView'
  var headerView: MDCFlexibleHeaderView?
                  ^~~~~~~~~~~~~~~~~~~~~
components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift:21:17: error: method does not override any method from its superclass
  override func viewDidLoad() {
  ~~~~~~~~      ^
```

**Review by romoore — COMMENTED** [review]
If possible, please break this PR up into two PRs. It will make it easier for reviewers to focus on each change.

1. Factoring out SimpleComposedTableViewController and SimpleInheritedTableViewController, then updating current examples
2. Adding the new example VC

File: components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift (hunk 1 above)
Comment (brianjmoore) [reply components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift]: Done. IDK why my Xcode was cool with missing imports, but it was.

**Comment by brianjmoore** [comment]
Split and reordered. This depends on https://github.com/material-components/material-components-ios/pull/6974 now, but IDK how to tell GitHub that to make the diff nice. This one really only adds one file.

File: components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift (hunk 1 above)
Comment (jverkoey) [reply components/AppBar/examples/supplemental/SimpleComposedTableViewController.swift]: bazel is much pickier about enforcing import statements than Xcode + CocoaPods, so these errors unfortunately only pop up in our CI or from a local run of bazel build.

**Comment by jverkoey** [comment]
> but IDK how to tell GitHub that to make the diff nice. This one really only adds one file.

We have a "PR:stacked" GitHub label we use to indicate that a PR contains another PR, feel free to use that 'til the other PR lands. It unfortunately doesn't solve the problem of showing separate PR contents, but it does help reviewers understand the PR is a composite.

**Comment by brianjmoore** [comment]
Other one is in - this diff is clean now and ready for review.

**Review by jverkoey — APPROVED** [review]

**Review by jverkoey — APPROVED** [review]

**Pull request merged**